ci-operator/step-registry/upi/install/azure/upi-install-azure-commands.sh

```diff
@@ -242,37 +282,59 @@ GATHER_BOOTSTRAP_ARGS="${GATHER_BOOTSTRAP_ARGS} --master ${MASTER0_IP} --master
 echo "Deploying 06_workers"
 WORKER_IGNITION=$(cat ${ARTIFACT_DIR}/installer/worker.ign | base64 -w0)
 export WORKER_IGNITION
-az deployment group create -g $RESOURCE_GROUP \
-  --template-file "06_workers.json" \
-  --parameters workerIgnition="$WORKER_IGNITION" \
-  --parameters sshKeyData="$SSH_PUB_KEY" \
-  --parameters baseName="$INFRA_ID"
+cmd="az deployment group create -g ${RESOURCE_GROUP} \
+    --template-file '06_workers.json' \
+    --parameters workerIgnition='${WORKER_IGNITION}' \
+    --parameters sshKeyData='${SSH_PUB_KEY}' \
+    --parameters baseName='${INFRA_ID}'"
+if [ -v vnet_basename ] && [ -n "${vnet_basename}" ]; then
+  echo "Using an existing vnet..."
+  cmd="${cmd} --parameters vnetBaseName='${vnet_basename}'"
+fi
+eval "${cmd}"
 
 popd
 echo "Waiting for bootstrap to complete"
 openshift-install --dir=${ARTIFACT_DIR}/installer wait-for bootstrap-complete &
 wait "$!" || gather_bootstrap_and_fail
 
 echo "Bootstrap complete, destroying bootstrap resources"
-az network nsg rule delete -g $RESOURCE_GROUP --nsg-name ${INFRA_ID}-nsg --name bootstrap_ssh_in
-az vm stop -g $RESOURCE_GROUP --name ${INFRA_ID}-bootstrap
+if [ -v vnet_basename ] && [ -n "${vnet_basename}" ]; then
+  az network nsg rule delete -g $RESOURCE_GROUP --nsg-name ${vnet_basename}-nsg --name bootstrap_ssh_in
+else
+  az network nsg rule delete -g $RESOURCE_GROUP --nsg-name ${INFRA_ID}-nsg --name bootstrap_ssh_in
+fi
```

Comment:
Instead of using a conditional here, can we set a nsg name variable earlier and use it?

Reply:
Done